design-system/atoms/input/adapters/default/default.adapter.tsx

```diff
@@ -0,0 +1,230 @@
+  const slots = InputDefaultVariants(variants);
+  const hasStartContent = startContent || startIcon || avatar;
+  const avatarSize: Record<NonNullable<typeof size>, ComponentProps<typeof Avatar>["size"]> = {
+    sm: "xxs",
+    md: "xs",
+    lg: "s",
+  };
+
+  if (!hasStartContent) {
+    return null;
+  }
+
+  return (
+    <div className={cn(slots.contentWrapper())}>
+      <div className={"flex flex-row items-center justify-start gap-1"}>
+        {startContent}
+        {startIcon ? <Icon size={"sm"} {...startIcon} /> : null}
+        {avatar ? <Avatar size={avatarSize[size]} {...avatar} /> : null}
+      </div>
+    </div>
+  );
+}
+
+function EndContent({
+  endIcon,
+  endContent,
+  button,
+  ...variants
+}: Pick<InputPort, "button" | "endContent" | "endIcon" | "size" | "isFocused" | "isError" | "isDisabled">) {
+  const slots = InputDefaultVariants(variants);
+
+  const isInnerButton = useMemo(() => {
+    if (button) {
+      if (isButtonSolid(button)) {
+        return button.variant === "tertiary";
+      } else {
+        return true;
+      }
+    }
+
+    return false;
+  }, [button]);
+
+  const hasEndContent = endIcon || endContent || isInnerButton;
+
+  if (!hasEndContent) {
+    return null;
+  }
+
+  if (isInnerButton) {
+    return (
+      <div className={"px-1"}>
+        <Button {...button} isDisabled={variants?.isDisabled} />
+      </div>
+    );
+  }
+
+  return (
+    <div className={cn(slots.contentWrapper())}>
+      <div className={"flex flex-row items-center justify-end gap-1"}>
+        {endContent}
+        {endIcon ? <Icon size={"sm"} {...endIcon} /> : null}
+      </div>
+    </div>
+  );
+}
+
+export const InputDefaultAdapter = forwardRef(function InputDefaultAdapter(
+  {
+    id,
+    name,
+    classNames,
+    isError,
+    isDisabled,
+    value,
+    onChange,
+    startContent,
+    startIcon,
+    avatar,
+    endContent,
+    info,
+    error,
+    description,
+    endIcon,
+    label,
+    placeholder,
+    canInteract,
+    button,
+    attr = {},
+    size = defaultSize,
+    isFocused: _isFocused,
+  }: InputPort,
+  ref: ForwardedRef<HTMLInputElement>
+) {
+  const [isFocused, setIsFocused] = useState(false);
+  const isOuterButton = useMemo(() => {
+    if (button) {
+      if (isButtonSolid(button)) {
+        return button.variant === "secondary";
+      }
+    }
+
+    return false;
+  }, [button]);
+
+  const variants = { isFocused: isFocused || _isFocused, size, isDisabled, isError, asOuterElement: isOuterButton };
+  const slots = InputDefaultVariants(variants);
+
+  function onFocus() {
+    setIsFocused(true);
+  }
+
+  function onBlur() {
+    setIsFocused(false);
+  }
+
+  function handleChanges(e: ChangeEvent<HTMLInputElement>) {
+    if (onChange) {
+      onChange(e);
+    }
+  }
+
+  return (
+    <div className={cn(slots.container(), classNames?.container)} {...attr}>
```

Comment:
Is the spread of generic attributes necessary ?

Reply:
as the focus state can be use in other case in the design system like is done on checkbox and checkbox button the component should provide a way to apply a focus / hover state outside off the component